lib/mayaUsd/fileio/utils/meshWriteUtils.h

```diff
@@ -29,11 +29,75 @@
 #include <pxr/usd/usdGeom/mesh.h>
 #include <pxr/usd/usdUtils/sparseValueWriter.h>
 
+#include <maya/MBoundingBox.h>
 #include <maya/MDagPath.h>
 #include <maya/MFnMesh.h>
 #include <maya/MObject.h>
 #include <maya/MString.h>
 
+/**
+ * Gets the minimum unique name of a DAG node.
+ *
+ * @param node  The node to find the name of.
+ *
+ * @return      The name as a Maya string.
+ */
+MString mayaGetUniqueNameOfDAGNode(const MObject& node);
+
+/**
+ * Finds a child plug with the given `name`.
+ *
+ * @param parent    The parent plug to start the search from.
+ * @param name      The name of the child plug to find. This should be the short name.
+ *
+ * @return          The plug if it can be found, or a null `MPlug` otherwise.
+ */
+MPlug mayaFindChildPlugWithName(const MPlug& parent, const MString& name);
+
+/**
+ * Finds a skinCluster directly connected upstream in the DG to the given mesh.
+ *
+ * @param mesh              The mesh to search from.
+ * @param skinCluster       Storage for the result. If no skinCluster can be found,
+ *                          this will be a null `MObject`.
+ *
+ * @return                  `MStatus::kSuccess` if the operation completed successfully.
+ */
+MStatus mayaGetSkinClusterConnectedToMesh(const MObject& mesh, MObject& skinCluster);
+
+/**
+ * Similar to `mayaGetSkinClusterConnectedToMesh`, except that instead of finding a
+ * directly-connected skinCluster, it searches the DG upstream of the mesh for any
+ * other skinClusters as well.
+ *
+ * @param mesh              The mesh to start the search from.
+ * @param skinClusters      Storage for the result.
+ *
+ * @return                  `MStatus::kSuccess` if the operation completed successfully.
+ */
+MStatus mayaGetSkinClustersUpstreamOfMesh(const MObject& mesh, MObjectArray& skinClusters);
```

Comment:
That's very possible, since MSVC symbol visibility is private by default and the other way round on macOS/Linux. Unfortunately I don't have a Windows machine at work to cross-compile against. Though I wonder...shouldn't the CI system on here have caught this when building against MSVC?

Reply:
Agreed, I'm surprised by this one. Once a change is ready to merge, we always run a few more checks to cover configurations that are not yet handled by PF. Maybe we would see it there.

Reply:
Addressed in 6460336.